analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,150 @@
+	posthoc[,"Pathway"] <- pathwayName;
+	posthoc[,"ClinVar"] <- clinVarName;
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc);
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc);
+}
+
+##############################
+#Analysis by Disease
+##############################
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology");
+diseaseType <- do.call("rbind", diseaseType)
+
+#Filter to significant entries
+diseaseTypeFilt <- diseaseType[diseaseType[,"AOV_P"]<0.01,]
+diseaseTypeFilt <- diseaseType[diseaseType[,"p.adj"]<0.05,] 
+
+#Now make all comparisons the same direction & write out results
+diseaseTypeFilt[,"DiseaseX"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarX"]), as.character(diseaseTypeFilt[,"VarY"]))
+diseaseTypeFilt[,"DiseaseY"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarY"]), as.character(diseaseTypeFilt[,"VarX"]))
+diseaseTypeFilt <- diseaseTypeFilt[-1:-2]
+diseaseTypeFilt <- diseaseTypeFilt[, c("DiseaseX", "DiseaseY", colnames(diseaseTypeFilt)[1:8])]
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#Now get diseases highly associated with a certain pathway
+diseaseTableTmp <- data.frame(table(diseaseTypeFilt[,c("DiseaseX", "Pathway")]));
+diseaseTableTmp <- diseaseTableTmp[diseaseTableTmp[,"Freq"]>(length(rownames(geneSetExpMat))*.25),]
+diseaseTableTmp <- diseaseTableTmp[order(-diseaseTableTmp[,"Freq"]),]
+
+#Now set up to create heatmaps
+keepPathways <- unique(diseaseTableTmp[,"Pathway"]);
+keepDisease <- unique(diseaseTableTmp[,"DiseaseX"]);
+tmpClinData <- clinData[c("short_histology", "broad_histology")];
+colnames(tmpClinData)[1] <- "Histology" 
+rownames(geneSetExpMat) <- gsub("HALLMARK", "", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- gsub("_", " ", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- trimws(rownames(geneSetExpMat))
+
+#Heatmap across all samples, only significant pathways
+if(length(keepPathways)>10)
+{
+png("plots/HeatmapPathwaysGenes_all.png", width=1080, height=720)
+pheatmap(geneSetExpMat[keepPathways,],
+border_color="black",
+color=inferno(length(geneSetExpMat) - 1), 
+annotation_col=tmpClinData[c("Histology")],
+show_colnames=F)
+dev.off()
+
+#Heatmap across significant histologies, and pathways
+tmpClinDataTmp  <- tmpClinData[tmpClinData[,"Histology"]%in%keepDisease,]
+tmpGeneSetMat <- geneSetExpMat[keepPathways,rownames(tmpClinDataTmp)]
+print(paste("The Filtered Gene Set Matrix has", nrow(tmpGeneSetMat), "rows"));
+print(paste("The Filtered Gene Set Matrix has", ncol(tmpGeneSetMat), "columns"));
+png("plots/HeatmapPathwaysGenes_Sig.png", width=1080, height=720)
+pheatmap(tmpGeneSetMat,
+border_color="black",
+color=inferno(length(geneSetExpMat) - 1), 
+annotation_col=tmpClinData[c("Histology")],
+show_colnames=F)
+dev.off()
+}
+
+if(length(keepPathways)<10)
```

Comment:
Oh right, it's because when you only have 16 samples you don't get a lot/any significant pathways in the analysis hence the filtered data frame has no rows.